src/receive_imf.rs

```diff
@@ -1693,14 +1687,6 @@ async fn apply_group_changes(
                 Some(reply_to) => rfc724_mid_exists(context, reply_to).await?.is_none(),
```

Comment:
So, as a quick fix, i'd check here that a message not only exists, but is also already downloaded

Reply:
And if it is not downloaded we just recreate the whole state as if we never saw that message?

Reply:
Yes, because we actually never saw it, we saw only Message-ID and some other headers, but not the membership changes it brings